### PR TITLE
add health endpoint

### PR DESCRIPTION
**Pull request by kasparg**
## What changes are proposed in this pull request?

Add health endpoint

## How is this patch tested?

(Details)

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [x] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by ivan-transferwise — APPROVED** [review]

**Comment by tomasatdatabricks** [comment]
Hey @kasparg , thanks for the contribution. Can you also add a test for the new endpoint and update documentation?

**Comment by andychow-db** [comment]
I've updated the description to indicate it is a new API. It'll require some documentation and information.

Secondly, if you're looking for health, I suggest using `/ping` rather than `/health`.

File: mlflow/server/__init__.py [hunk 1]
```diff
@@ -37,6 +37,12 @@ def serve_artifacts():
     return get_artifact_handler()
 
 
+# Add health endpoint
```
Comment (andychow-db) [inline mlflow/server/__init__.py]: Nit: `Add health endpoint` is not the correct comment here. Please update it.

**Review by gopalsi — APPROVED** [review]

**Comment by mparkhe** [comment]
hi @kasparg : as recommended by @tomasatdatabricks, can you please add a test for the new endpoint?

**Review by aarondav — CHANGES_REQUESTED** [review]
A minor test and update to the doc comment are requested!

**Comment by aarondav** [comment]
Looks like @crflynn addressed these comments in #2725, which was merged. Thanks!

**Pull request closed without merging**